### ISSUE TITLE
Registries don't meet Process requirements

### ISSUE THREAD

**Issue by nigelmegitt** [w3c/webcodecs#644]
The [Codec Registry](https://w3c.github.io/webcodecs/codec_registry.html) and the [VideoFrame Metadata Registry](https://w3c.github.io/webcodecs/video_frame_metadata_registry.html) don't meet the [registry requirements](https://www.w3.org/2021/Process-20211102/#registries) of the Process.

For example, they aren't showing as "Registry" as the document type, and they don't link to [§ 6.5 The Registry Track](https://www.w3.org/2021/Process-20211102/#registries) - that latter requirement is defined in [6.5.2. Publishing Registries](https://www.w3.org/2021/Process-20211102/#reg-pub).

I've summarised some of the requirements for Registries in the discussion at w3c/ttwg#241 and shared my thinking about registry definitions for TTWG specifications, which may be relevant here too, since there's not really anything timed-text related about them. See also the draft boilerplate text at w3c/ttwg#243 - at the very least that does the (not yet well documented) Respec thing needed to make the document look like a Registry track document.

**Comment by dalecurtis** [comment]
@tidoust @cpn do you have suggestions here? Editors have no idea what this means :)

**Comment by chrisn** [comment]
Thanks, yes, I'll review this when I'm back at work next week.

**Comment by tidoust** [comment]
> For example, they aren't showing as "Registry" as the document type

You linked to the Editor's Draft of the registry, not to the published version, e.g. [WebCodecs Codec Registry](https://www.w3.org/TR/webcodecs-codec-registry/). The published version has the right "Draft Registry" type. The Editor's Draft is just a draft. There is no specific status in ReSpec and Bikeshed to identify the non-published draft of a Registry document. Editor's Draft seems good enough to me.

> and they don't link to [§ 6.5 The Registry Track](https://www.w3.org/2021/Process-20211102/#registries)

I would argue that the sentence "This document was published by the [Media Working Group](https://www.w3.org/groups/wg/media) as a Draft Registry using the [Registry track](https://www.w3.org/2021/Process-20211102/#recs-and-notes)", in the [Status of This Document](https://www.w3.org/TR/webcodecs-codec-registry/#sotd) section, is intended to cover that requirement. That said, the link is to [$6.1 Types of Technical Reports](https://www.w3.org/2021/Process-20211102/#recs-and-notes) and could perhaps be updated to rather target $6.5, or the Process could perhaps be amended to be less strict about that requirement. If the former, the link cannot be updated without first updating PubRules. @deniak FYI.

**Comment by nigelmegitt** [comment]
> There is no specific status in ReSpec and Bikeshed to identify the non-published draft of a Registry document.

```
      var respecConfig = {
        specStatus: "RY",
```

works.



> That said, the link is to [$6.1 Types of Technical Reports](https://www.w3.org/2021/Process-20211102/#recs-and-notes) and could perhaps be updated to rather target $6.5, or the Process could perhaps be amended to be less strict about that requirement.

It does seem strange that Registry track documents are the only ones that are required to link to an explicit process section, but it _is_ very clear that it wants a link to §6.5 and not §6.1.

**Comment by tidoust** [comment]
The `"RY"` (or `"DRY"`) spec status is a good status for the version of a spec published under `https://www.w3.org/TR/` but the version published under `https://w3c.github.io/` is an Editor's Draft, and its status should reflect that. For specs on the Note track, [Bikeshed](https://speced.github.io/bikeshed/#metadata) has a `"NOTE-ED"` status (which does not exist in ReSpec as far as I can tell). Introducing new statuses in authoring tools for Editor's Drafts of a registry spec seems overkill though.

**Comment by nigelmegitt** [comment]
> Introducing new statuses in authoring tools for Editor's Drafts of a registry spec seems overkill though

I disagree, I think it seems appropriate.

**Comment by chrisn** [comment]
Is there a more appropriate home for this tooling discussion? This doesn't seem to be particularly WebCodecs specific (unless there are other changes needed to these registries?)

**Comment by tidoust** [comment]
> Is there a more appropriate home for this tooling discussion? This doesn't seem to be particularly WebCodecs specific (unless there are other changes needed to these registries?)

Yes, for cross authoring tools discussions, the spec-prod@w3.org mailing-list (with [public archives](https://lists.w3.org/Archives/Public/spec-prod/)) would be the most appropriate place.

For ReSpec: https://github.com/w3c/respec/issues.
For Bikeshed: https://github.com/speced/bikeshed/issues

**Comment by padenot** [comment]
Is there anything we should do as editors here? We still have no idea what all of this means, but we're happy to tag our documents appropriately, of course.

**Comment by nigelmegitt** [comment]
@padenot yes, you need to include the required reference to [Registry track](https://www.w3.org/2021/Process-20211102/#recs-and-notes) . For an example, see the Respec generated SOTD text at https://ttml.w3c.himor.in/boilerplate-registry-20230316.html

**Comment by chrisn** [comment]
I have filed https://github.com/w3c/w3process/issues/725 to ask whether there's a real need to link to the Process at all, as this isn't required for other types of report.

@deniak has filed https://github.com/w3c/specberus/issues/1713 to update the link to point to [6.5](https://www.w3.org/2021/Process-20211102/#registries) rather than [6.1](https://www.w3.org/2021/Process-20211102/#recs-and-notes) .

@tidoust already commented on whether `ED` is the appropriate status for the draft on [w3c.github.io](https://w3c.github.io/webcodecs/codec_registry.html). But I agree that the current status seems confusing. I see @nigelmegitt has filed https://github.com/w3c/respec/issues/3594 for Respec.

So I suggest waiting on the outcome of those conversations before making changes here.

**Comment by chrisn** [comment]
I'll re-open, as I think commit https://github.com/w3c/webcodecs/commit/d74a047f7605304eba25941b7bd331058612a6ff mistakenly linked to this issue.

That said, https://www.w3.org/TR/webcodecs-codec-registry/ does now have the link to the Registry Track. I'll check what's left to do to resolve this issue.

**Comment by aboba** [comment]
@chrisn Is there anything else we need to do, or can we close this?

**Comment by chrisn** [comment]
I think this is fine as is, if new draft document types get added to the tooling we can revisit, but as @tidoust said, the actual registry on /TR  has the appropriate links.
